lib/partials2melodies.js

```diff
@@ -29,21 +29,21 @@ function partials2melodies(_partials) {
 
     // define subfunctions
     function detectNoteOnNoteOffIndices(amps) {
-      const result = [];
+      const noteOnOffs = [];
       // 'note on' = 1, 'note off' = -1, 'nothing' = 0
       amps.forEach((amp, idx, amps) => {
         if (idx == 0) {
-          if (amp > 0) result.push(1);
-          else result.push(0);
+          if (amp > 0) noteOnOffs.push(1);
+          else noteOnOffs.push(0);
         } else if (amp > 0 && amps[idx - 1] == 0) {
-          result.push(1);
+          noteOnOffs.push(1);
         } else if (amp == 0 && amps[idx - 1] > 0) {
-          result.push(-1);
+          noteOnOffs.push(-1);
         } else {
-          result.push(0);
+          noteOnOffs.push(0);
         }
       });
-      return result;
+      return noteOnOffs;
     }
 
     function calcActiveNoteNums(freqs, noteOnOffs) {
```

Comment:
Maybe it's just me but,
I think this function name potentially involves two separate confusions.

1. `Nums` could mean both  **sum** or **indices**.
2. `calc` is still ambiguous. Maybe better replaced with something related to the operation.

For example, 
If I were to write a function that collects all the positions of active notes,
I would define `collectActiveNotePos`/`collectActiveNoteIndices`/`findAllActiveNotes`, etc.

If I were to write a function that gets the total number of active notes,
I would define `getSumOfActiveNotes` or something.

Reply:
However, 
@zigen is more practiced in programming, in general. 
Probably he could come up with more suitable names for these kind of situations.